addons/payment_authorize/models/payment.py

```diff
@@ -48,13 +48,36 @@ def _get_authorize_urls(self, environment):
             return {'authorize_form_url': 'https://test.authorize.net/gateway/transact.dll'}
 
     def _authorize_generate_hashing(self, values):
+        def is_signature_key(key):
+            ''' Try to detect if the key is new hexa signature key or the old
+            transaction key. Basically this method checks if the user proceeded
+            to the key change required from Authorize.net the 28th March 2019.
+
+            Transaction key (old): 16 length, anything between 0-9, a-z and A-Z.
+            Signature key (new): 128 length, hexa, only 0-9 and A-Z.
+            '''
+            try:
+                int(key, 16)
+                has_lower = [c for c in key if c.islower()]
+                if has_lower or len(key) <= 16:
+                    return False
+            except ValueError:
+                return False
+            return True
+
         data = '^'.join([
             values['x_login'],
             values['x_fp_sequence'],
             values['x_fp_timestamp'],
             values['x_amount'],
             values['x_currency_code']])
-        return hmac.new(str(values['x_trans_key']), data, hashlib.md5).hexdigest()
+
+        if is_signature_key(values['x_trans_key']):
+            # [BACKWARD COMPATIBILITY] Check that the merchant did update his transaction
```

Comment:
comment is ambiguous, you have the impression that the backward compatibility is the first if, while it is the else.

Reply:
Done, comment is before the `if`, makes more sense